### ISSUE TITLE
Adding a third or more column does not show the name in attribute table

### ISSUE THREAD

**Issue by deltamarnix** [qgis/QGIS#55533]
### What is the bug or the crash?

The moment that you create a new layer and add a number of columns in the attribute table, from the third column onwards, the names don't show up.

### Steps to reproduce the issue

1. Open QGIS
2. Add a shape file (name: foo, no geometry)
3. Open Attribute Table
4. Toggle Editing Mode
5. Add new field (foo)
6. Add new field (bar) **There is a column for bar, but the word itself is not displayed**
7. Toggle editing mode, save layers
8. Close and reopen the attribute table in order to see all names

### Versions

3.34.0

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by agiudiceandrea** [comment]
Hi @deltamarnix, the issue about the incorrect hiding of columns in an empty vector layer has been already previously reported at https://github.com/qgis/QGIS/issues/49142.
Could you please open a new issue report about the other reported issue (new field name not displayed in QGIS 3.34) or change the title and modify the description of this issue report accordingly?

**Comment by deltamarnix** [comment]
Thanks, I have simplified this bug report.

**Comment by nirvn** [comment]
@deltamarnix , I'm failing to grasp your issue, maybe you could add a couple of screenshots?

**Comment by deltamarnix** [comment]
This is during step 6

![image](https://github.com/qgis/QGIS/assets/150045289/84004c2e-f80a-4b9d-9ec7-8648131617de)

This is after step 8

![image](https://github.com/qgis/QGIS/assets/150045289/5dff2be1-6237-430c-9725-87026cfad5f7)

**Comment by agiudiceandrea** [comment]
@nirvn here is a screen recording:


https://github.com/qgis/QGIS/assets/16253859/6fd5655e-3165-4612-953e-66ac89459a53

**Comment by agiudiceandrea** [comment]
@nirvn here is a screen recording:

https://github.com/qgis/QGIS/assets/16253859/18bc4e3f-0dec-4879-9577-e8c273883f68
